src/platform/graphics/egl_extensions.cpp

```diff
@@ -56,30 +45,37 @@ std::experimental::optional<mg::EGLExtensions::PlatformBaseEXT> maybe_platform_b
 }
 
 mg::EGLExtensions::EGLExtensions() :
+    platform_base{maybe_platform_base_ext()}
+{
+}
+
+mg::EGLExtensions::BaseExtensions::BaseExtensions(EGLDisplay dpy) :
     eglCreateImageKHR{
         reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(eglGetProcAddress("eglCreateImageKHR"))},
     eglDestroyImageKHR{
         reinterpret_cast<PFNEGLDESTROYIMAGEKHRPROC>(eglGetProcAddress("eglDestroyImageKHR"))},
-
     /*
      * TODO: Find a non-ES GL equivalent for glEGLImageTargetTexture2DOES
      * It's the LAST remaining ES-specific function. Although Mesa lets you use
      * it in full GL, it theoretically should not work. Mesa just lets you
      * mix ES and GL code. But other drivers won't be so lenient.
      */
     glEGLImageTargetTexture2DOES{
-        reinterpret_cast<PFNGLEGLIMAGETARGETTEXTURE2DOESPROC>(eglGetProcAddress("glEGLImageTargetTexture2DOES"))},
-    wayland{maybe_wayland_ext()},
-    platform_base{maybe_platform_base_ext()}
+        reinterpret_cast<PFNGLEGLIMAGETARGETTEXTURE2DOESPROC>(eglGetProcAddress("glEGLImageTargetTexture2DOES"))}
 {
-    if (!eglCreateImageKHR || !eglDestroyImageKHR)
-        BOOST_THROW_EXCEPTION(std::runtime_error("EGL implementation doesn't support EGLImage"));
+    auto const egl_extensions = eglQueryString(dpy, EGL_EXTENSIONS);
+    if (!egl_extensions || !strstr(egl_extensions, "EGL_KHR_image_base") || !eglCreateImageKHR || !eglDestroyImageKHR)
+    {
+        BOOST_THROW_EXCEPTION(std::runtime_error("EGL display doesn't support EGL_KHR_image_base"));
+    }
 
     if (!glEGLImageTargetTexture2DOES)
+    {
```

Comment:
This is still wrong; we're using the extension without checking for its presence. That said, this is a strict improvement on the status quo.

Reply:
Can you fix it then? I do not know what I'm supposed to do here.

Reply:
Fair enough. We need to move `glEGLImageTargetTexture2DOES` entirely out of `EGLExtensions`, as it's not an EGL extension, and instead have a `GLExtensions` object which gets constructed with an active GL context (as GL extensions are per *context*).

Reply:
I created #1808 so we wont loose it if we don't deal with it now.

Reply:
Ta